packages/scan/src/ScanFactory .ts

```diff
@@ -0,0 +1,90 @@
+import { Discovery, Module } from './enums';
+import { ScanConfig, Scans, ScanSettings, Target } from './Scans';
+import { Scan } from './Scan';
+import { Har, HarEntryBuilder } from './HarEntryBuilder';
+import { Configuration } from '@secbox/core';
+import { v4 } from 'uuid';
+
+export class ScanFactory {
+  private readonly scans: Scans;
+
+  constructor(private readonly sdkConfig: Configuration) {
+    this.scans = sdkConfig.container.resolve(Scans);
+  }
+
+  public async createScan(settings: ScanSettings): Promise<Scan> {
+    const scanConfig: ScanConfig = await this.buildScanConfig(settings);
+
+    const { id } = await this.scans.create(scanConfig);
+
+    return new Scan(id, this.scans);
+  }
+
+  private async buildScanConfig(settings: ScanSettings): Promise<ScanConfig> {
+    const fileId = await this.createFile(settings.target);
+
+    return {
+      fileId,
+      module: Module.DAST,
+      name: settings.name,
+      smart: settings.smart,
+      tests: settings.tests,
+      poolSize: settings.poolSize,
+      discoveryTypes: [Discovery.CRAWLER],
```

Comment:
You should use only `Discovery.ARCHIVE`. It implies that you should pass `fileId` instead of `crawlerUrls`